include/f_core/net/device/c_lora.h

```diff
@@ -0,0 +1,81 @@
+#ifndef C_LORA_H
+#define C_LORA_H
+
+#include <zephyr/drivers/lora.h>
+
+class CLora {
+public:
+    /**
+     * Constructor
+     * @param lora_dev[in] The LoRa device to use
+     */
+    explicit CLora(const device& lora_dev);
+
+    /**
+     * Constructor
+     * @param lora_dev[in] The LoRa device to use
+     * @param config[in] Configuration for the LoRa modem
+     */
+    explicit CLora(const device& lora_dev, const lora_modem_config& config);
+
+    /**
+     * Synchronously transmit data (blocking)
+     * @param data[in] Data to transmit
+     * @param len[in] Size of data
+     * @return Zephyr status code
+     */
+    int TransmitSynchronous(const void* data, size_t len);
+
+    /**
+     * Receive data synchronously (blocking)
+     * @param data[out] Buffer to receive data into
+     * @param len[out] Size of buffer for receiving data
+     * @param rssi[in] Pointer to received signal strength indicator from received packet
+     * @param snr[in] Pointer to signal to noise ratio from received packet
+     * @param timeout[out] Timeout for receiving data
+     * @return Zephyr status code
+     */
+    int ReceiveSynchronous(void* data, size_t len, int16_t *rssi, int8_t *snr, k_timeout_t timeout = K_FOREVER);
+
+    /**
+     * Transmit data asynchronously (non-blocking)
+     * @param data[in] Data to transmit
+     * @param len[in] Size of data
+     * @param signal[in] Signal to notify when transmission is complete
+     * @return Zephyr status code
+     */
+    int TransmitAsynchronous(const void* data, size_t len, k_poll_signal *signal);
+
+    /**
+     * Receive data asynchronously (non-blocking)
+     * @param cb Callback to run when a packet is received
+     * @return Zephyr status code
+     */
+    int ReceiveAsynchronous(lora_recv_cb cb);
+
+    // TODO: Implement configuration functions
+
+private:
+    const device* lora_dev;
```

Comment:
Usually they should not change. In terms of IREC, our frequency should be fixed. Usually our bandwidth, spreading factor, etc should not need to change either. This would be a nice to have though if necessary. I do want a function that (unsafely) sets the bandwidth though. This would involve getting the parent SPI bus, determining the chipset and setting the registers according to chipset. This is because Zephyr's implementation limits us to 125 kHz of bandwidth and we can't go any lower which limits our range.

Reply:
Configuration is left up to the user in application land